exercises/prime-factors/canonical-data.json

```diff
@@ -0,0 +1,62 @@
+{
+  "for": {
+    "description": "returns prime factors for the given input number",
```

Comment:
The format that I was following (from other exercises) had the "method name" as the key for a block,  then a description for what that method did, then the cases for that method call tests.

The examples I was following are here:
* https://github.com/exercism/x-common/tree/master/exercises/allergies/canonical-data.json
* https://github.com/exercism/x-common/tree/master/exercises/alphametics/canonical-data.json
* https://github.com/exercism/x-common/tree/master/exercises/atbash-cipher/canonical-data.json
* https://github.com/exercism/x-common/tree/master/exercises/binary/canonical-data.json
* https://github.com/exercism/x-common/tree/master/exercises/book-store/canonical-data.json
* https://github.com/exercism/x-common/tree/master/exercises/bowling/canonical-data.json
* https://github.com/exercism/x-common/tree/master/exercises/change/canonical-data.json
* https://github.com/exercism/x-common/tree/master/exercises/clock/canonical-data.json**
* https://github.com/exercism/x-common/tree/master/exercises/luhn/canonical-data.json

Somewhere in my travels through all different tickets and discussions around canonical data, I thought I read something that suggested a preference for the method name being included (not just the `cases` key).  So that's what that is for.